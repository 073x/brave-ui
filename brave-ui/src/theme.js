/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

const theme = {
  browserButton: {
    default: {
      bg: '#fff',
      color: '#5a5a5a',
      hoverColor: '#000'
    },
    primary: {
      bg: 'linear-gradient(#ff7a1d, #ff5000)',
      gradient1: '#ff7a1d',
      gradient2: '#ff5000',
      color: '#fff',
      hoverColor: '#fff',
      borderHoverColor: '#fff'
    },
    secondary: {
      bg: 'linear-gradient(#fff, #ececec)',
      color: '#666',
      hoverColor: 'rgb(153, 153, 153)',
      borderHoverColor: '#444'
    }
  },
  switchButton: {
    checkedBg: '#ff6000',
    uncheckedBg: '#ccc',
    knob: '#fff'
  }
}

export default theme
