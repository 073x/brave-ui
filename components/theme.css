:root {
  /* theme for browserButton */
  /* -> browserButton default color */
  --browserButton-default--bg: #fff;
  --browserButton-default--color: #5a5a5a;
  --browserButton-default--hoverColor: #000;

  /* -> browserButton primary color */
  --browserButton-primary--bg: linear-gradient(#ff7a1d, #ff5000);
  --browserButton-primary--gradient1: #ff7a1d;
  --browserButton-primary--gradient2: #ff5000;
  --browserButton-primary--color: #fff;
  --browserButton-primary--hoverColor: #fff;
  --browserButton-primary--borderHoverColor: #fff;

  /* -> browserButton secondary color */
  --browserButton-secondary--bg: linear-gradient(#fff, #ececec);
  --browserButton-secondary--color: #666;
  --browserButton-secondary--hoverColor: rgb(153, 153, 153);
  --browserButton-secondary--borderHoverColor: #444;

  /* theme for switchButton */
  --switchButton-unchecked--bg: #ccc;
  --switchButton-checked--bg: inset 0 0 0 var(--height, 16px) #ff6000;
  --switchButton-knob--bg: #fff;
}
