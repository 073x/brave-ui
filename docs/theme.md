# Theme properties

We use a central list of variables for some values, such as color, fonts, spacing for the following purposes:
- Consistency
- Variability (e.g. Dark vs Light)
- Ease of change

## Using theme properties
Every styled-components Component will have access to a `theme` prop. This is because all Views are wrapped with a `<ThemeProvider theme={x}>` Component.

Checkout the list of theme properties at `theme-interface.tsx`.

## Adding a Theme Property
Properties in the main theme interface should be general and easy to understand the purpose for. Therefore, be careful not to add things that are only in use for a single control.

E.g. a suitable color property would be `primaryBackground` or `warningBackground` but an unsuitable one would be `buttonBackground`.

E.g. a suitable font-size strategy would be to define a scale of small to large sizes, such as `fontSizes: ['8', '12', '18']` rather than `buttonFontSizeLarge: '18'`

All new theme properties must be added to the theme interface, as well as values for that property defined for each theme.
