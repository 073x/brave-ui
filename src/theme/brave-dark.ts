import ITheme from './theme-interface'
import colors from './colors'
import defaultTheme from './brave-default'

const darkTheme: ITheme = {
  ...defaultTheme,
  name: 'Brave Dark',
  color: {
    ...defaultTheme.color,
    contextMenuBackground: colors.black,
    contextMenuForeground: colors.white,
    defaultControl: colors.grey400,
    defaultControlInteracting: colors.white,
    defaultControlActive: colors.grey500
  }
}

export default darkTheme
