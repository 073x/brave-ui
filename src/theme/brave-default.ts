import colors from './palette'
import ITheme from './theme-interface'

// Define colors, fonts, and sizes by purpose.
// Keys should not describe the value, but what they are to be used for.

const theme: ITheme = {
  name: 'Brave Light',
  color: {
    // Brand Primary
    primary: colors.orange500,
    primaryHover: colors.orange400,
    primaryActive: colors.orange600,

    // Brand Accent
    accent: colors.blurple500,
    accentHover: colors.blurple400,
    accentActive: colors.blurple600,

    // Brand Destructive
    destructive: colors.red500,
    destructiveHover: colors.red400,
    destructiveActive: colors.red600,

    // Brand Warn
    warn: colors.yellow500,
    warnHover: colors.yellow400,
    warnActive: colors.yellow600,

    // Brand Error
    error: colors.red500,
    errorHover: colors.red400,
    errorActive: colors.red600,

    // Brand Info
    info: colors.blue500,
    infoHover: colors.blue400,
    infoActive: colors.blue600,

    // Brand Success
    success: colors.green500,
    successHover: colors.green400,
    successActive: colors.green600,

    // Brand Grey
    grey000: colors.grey000,
    grey100: colors.grey100,
    grey200: colors.grey200,
    grey300: colors.grey300,
    grey400: colors.grey400,
    grey500: colors.grey500,
    grey600: colors.grey600,
    grey700: colors.grey700,
    grey800: colors.grey800,
    grey900: colors.grey900,

    // Backgrounds
    bodyBackground: colors.white,
    componentBackground: colors.white,
    componentAltBackground: colors.grey200,

    // Modal
    modalOverlayBackground: 'rgba(36,37,54,0.85)',

    // Typography
    headingDefault: colors.grey800, // default heading color
    textDefault: colors.grey700, // default text color
    textSecondary: colors.grey500,
    textSubtle: colors.grey200,

    // Toggle
    toggleSlider: colors.grey300, // other elements are tied to brandPrimary

    // Radio
    radioStroke: colors.grey300, // other elements are tied to brandPrimary,

    // Button
    btnDisabled: colors.grey300, // other elements are tied to brandPrimary,

    disabledAlpha: '.50' // the transparency for disabled components
  },
  fontSize: {
    default: '14px',
    large: '16px',
    small: '12px'
  },
  lineHeight: {
    default: '1.5',
    large: '1.75',
    none: '0' // neccessary?
  },
  spacing: {
    // Spacing
    extraLarge: '32px',
    large: '24px',
    medium: '16px',
    small: '12px',
    extraSmall: '8px',
    tiny: '4px'
  },
  radius: {
    default: '8px', // default for most elements
    small: '4px', // used for smaller components - inputs/forms
    circle: '50%' // used to create half circles on elements - buttons
  },
  border: {
    default: {
      color: colors.grey300, // base border outline a component
      size: '1px', // width of the border for a component
      style: 'solid' // style of a components border
    }
  },
  animation: {
    transitionDurationSlow: '0.6s',
    transitionDurationBase: '0.4s',
    transitionDurationFast: '0.2s'
  },
  shadow: {
    large: '0 4px 12px rgba(0, 0, 0, 0.2)',
    default: '0 2px 8px rgba(0, 0, 0, 0.2)',
    small: '0 2px 2px rgba(0, 0, 0, 0.2)'
  },
  fontFamily: {
    heading: 'Poppins, sans-serif',
    body: 'Muli, sans-serif',
    code: 'SFMono-Regular, monospace',
    system:
  '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif'
  }
}

export default theme
