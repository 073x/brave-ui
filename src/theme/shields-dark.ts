import ITheme from './theme-interface'
import defaultTheme from './brave-default'
import colors from './colors'

const shieldsDarkTheme: ITheme = {
  ...defaultTheme,
  name: 'Shields Dark',
  textColor: colors.white,
  panelBackground: colors.grey900,
  inputBorder: colors.grey700,
  separatorLine: colors.grey800
}

export default shieldsDarkTheme
