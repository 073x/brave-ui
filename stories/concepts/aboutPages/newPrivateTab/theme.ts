/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

const theme = {
  page: {
    fontFamily: '"Poppins", sans-serif',
    color: '#fff',
    padding: '80px 60px 40px',
    background: 'linear-gradient(#4b3c6e, #000)'
  },
  clock: {
    color: 'rgba(255,255,255,0.8)'
  },
  trackersBlocked: {
    dataItem: { color: '#f39030' }
  },
  adsBlocked: {
    dataItem: {color: '#fe521d' }
  },
  httpsUpgrades: {
    dataItem: {color: '#0796fa' }
  },
  estimatedTime: {
    dataItem: {color: '#999999' }
  },
  // todo
  clockContainer: {
    column: {
      justifyContent: 'flex-end'
    }
  },
  textualBlock: {
    boxedContent: {
      maxWidth: '650px',
      margin: '60px auto 0',
      fontFamily: '"Muli", sans-serif'
    }
  },
  media: {
    mediaContent: {
      width: '80px',
      margin: '0 0 0 25px'
    }
  },
  title: {
    heading: {
      fontFamily: '"Poppins", sans-serif',
      fontWeight: '400',
      color: '#fff'
    }
  },
  italicText: {
    paragraph: {
      fontStyle: 'italic',
      fontSize: '15px'
    }
  },
  text: {
    paragraph: {
      fontSize: '18px'
    }
  },
  switchContainer: {
    boxedContent: {
      margin: '40px 0 25px'
    }
  }
}

export default theme
