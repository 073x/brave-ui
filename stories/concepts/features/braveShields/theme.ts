/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

const theme = {
  braveShieldsHeader: {
    backgroundColor: '#808080',
    padding: '10px',
    gridGap: '0',
    color: '#fafafa'
  },
  columnVerticalCenter: {
    alignItems: 'center'
  },
  columnVerticalCenterEnd: {
    alignItems: 'center',
    justifyContent: 'flex-end'
  },
  title: {
    fontSize: '14px'
  },
  closeButton: {
    fontSize: '20px'
  },
  hostnameContent: {
    flexDirection: 'column'
  },
  hostname: {
    fontSize: '20px'
  },
  braveShieldsControls: {
    backgroundColor: '#eee',
    padding: '10px 10px 0',
    gridGap: '10px 5px'
  },
  braveShieldsControlsContent: {
    margin: '0 5px'
  },
  braveShieldsControlsSwitches: {
    gridGap: '10px 5px',
    padding: '5px 0'
  },
  braveShieldsFooter: {
    gridGap: '10px',
    padding: '0 10px 10px',
    backgroundColor: '#eee'
  },
  columnStart: {
    justifyContent: 'flex-start'
  },
  columnEnd: {
    justifyContent: 'flex-end'
  },
  editLink: {
    textDecoration: 'none'
  },
  braveShieldsStats: {
    padding: '10px',
    gridGap: '7px',
    backgroundColor: '#f7f7f7'
  },
  adsTrackersBlockedStats: {
    userSelect: 'none',
    counterColor: '#fe521d',
    descriptionColor: '#3b3b3e'
  },
  httpsUpgradesStats: {
    userSelect: 'none',
    counterColor: '#0796fa',
    descriptionColor: '#3b3b3e'
  },
  scriptsBlockedStats: {
    userSelect: 'none',
    counterColor: '#555555',
    descriptionColor: '#3b3b3e'
  },
  fingerPrintingProtectionStats: {
    userSelect: 'none',
    counterColor: '#ffc000',
    descriptionColor: '#3b3b3e'
  }
}

export default theme
