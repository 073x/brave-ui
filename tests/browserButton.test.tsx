/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import BrowserButton from '../brave-ui/src/browserButton'
import theme from '../brave-ui/src/theme'

describe('browserButton tests', () => {
  it('matches the snapshot', () => {
    const component = <BrowserButton />
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component', () => {
    const wrapper = shallow(<BrowserButton id='browserButton' />)
    const assertion = wrapper.find('#browserButton').length
    expect(assertion).toBe(1)
  })

  it('defines a custom size', () => {
    const wrapper = shallow(<BrowserButton size='13px' />)
    const assertion = wrapper.find('button').props().style
    expect(assertion).toEqual(expect.objectContaining({ '--size': '13px' }))
  })

  it('defines a custom font size', () => {
    const wrapper = shallow(<BrowserButton fontSize='60px' />)
    const assertion = wrapper.find('button').props().style
    expect(assertion).toEqual(expect.objectContaining({ '--fontSize': '60px' }))
  })

  it('can pass text', () => {
    const wrapper = shallow(<BrowserButton>NESPRESSO</BrowserButton>)
    const assertion = wrapper.find('button').text()
    expect(assertion).toBe('NESPRESSO')
  })

  it('can be disabled', () => {
    const wrapper = shallow(<BrowserButton disabled />)
    const assertion = wrapper.find('button').props().disabled
    expect(assertion).toBe(true)
  })

  it('defaults to the default button', () => {
    const wrapper = shallow(<BrowserButton />)
    const assertion = wrapper.find('button').props().style
    expect(assertion).toEqual(
      expect.objectContaining({ '--bg': theme.browserButton.default.bg })
    )
  })

  it('can be set as primary', () => {
    const wrapper = shallow(<BrowserButton theme='primary' />)
    const assertion = wrapper.find('button').props().style
    expect(assertion).toEqual(
      expect.objectContaining({ '--primary-bg': theme.browserButton.primary.bg })
    )
  })

  it('can be set as secondary', () => {
    const wrapper = shallow(<BrowserButton theme='secondary' />)
    const assertion = wrapper.find('button').props().style
    expect(assertion).toEqual(
      expect.objectContaining({ '--secondary-bg': theme.browserButton.secondary.bg })
    )
  })

  it('can respond to click', () => {
    const value = 'FANCY BUTTON CLICKED'
    const onClick = jest.fn()
    const wrapper = shallow(<BrowserButton onClick={onClick} />)
    wrapper.find('button').simulate('click', value)
    expect(onClick).toBeCalledWith(value)
  })
})
