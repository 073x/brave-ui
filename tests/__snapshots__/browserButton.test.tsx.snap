// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`browserButton tests matches the snapshot 1`] = `
<button
  className="0"
  disabled={undefined}
  id={undefined}
  onClick={undefined}
  style={
    Object {
      "--fontSize": undefined,
      "--size": undefined,
    }
  }
/>
`;
